src/irr/asset/CMeshManipulator.cpp

```diff
@@ -718,6 +334,19 @@ asset::ICPUMeshBuffer* CMeshManipulator::createMeshBufferUniquePrimitives(asset:
 	return clone;
 }
 
+//
+asset::ICPUMeshBuffer* CMeshManipulator::calculateSmoothNormals(asset::ICPUMeshBuffer* inbuffer, float epsilon,
+	asset::E_VERTEX_ATTRIBUTE_ID normalAttrID, VxCmpFunction vxcmp) const
+{
+	if (inbuffer == nullptr)
+		return nullptr;
+
+	asset::ICPUMeshBuffer* outbuffer = createMeshBufferDuplicate(inbuffer);
```

Comment:
Maybe take heed from 
https://github.com/buildaworldnet/IrrlichtBAW/blob/839516417231e3a47dcb1781b9414965ca74a85b/include/irr/asset/IMeshManipulator.h#L105

because the mesh does not always need to be created as a duplicate (in-fact that can be harmful to other optimizations).